VelocitySDK/VLTClient.m

```diff
@@ -88,21 +90,34 @@ - (void)initializeWsApiClient
     self.wsApiClient = [[VLTWsApiClient alloc] initWithQueueSize:VLTWsApiQueueSize];
     [self.wsApiClient setOnClose:^(NSInteger code, NSString *_Nonnull reason) {
         DLog(@"WS Closed: %ld [%@]", (long)code, reason);
+        vlt_strongify(self);
+        if (code != SRStatusCodeNormal && code != SRStatusCodeGoingAway) {
+            [self reconnectAndStartMotionSensing];
+        }
     }];
     [self.wsApiClient setOnError:^(NSError *_Nonnull error) {
         DLog(@"WS error: %@", error);
         vlt_strongify(self);
-        [self reinitializeWsApiClient];
-        [self startMotionSensing];
+        [self reconnectAndStartMotionSensing];
     }];
     [self.wsApiClient setOnPong:^(NSData *_Nullable pongPayload) {
         DLog(@"WS API Client PONG: %@", [[NSString alloc] initWithData:pongPayload encoding:NSUTF8StringEncoding]);
     }];
 }
 
+- (void)reconnectAndStartMotionSensing
+{
+    [self reinitializeWsApiClient];
+    [self startMotionSensing];
+}
+
 - (void)closeWsApiClientIfNeeded
 {
     if (self.wsApiClient && !self.wsApiClient.isClosed) {
+        [self.wsApiClient setOnOpen:nil];
+        [self.wsApiClient setOnClose:nil];
+        [self.wsApiClient setOnError:nil];
+        [self.wsApiClient setOnPong:nil];
```

Comment:
Why are these needed? If the callback blocks doesn't retain variables these shouldn't be necessary?

Reply:
As there a time period between `close` and actual object deallocation, but in that we are not interesting into those blocks anymore. But maybe I can move those to `VLTWsApiClient` it self